django_project/core/session_views.py

```diff
@@ -0,0 +1,45 @@
+from rest_framework import viewsets
+from rest_framework.response import Response
+from core.models import UserSession
+from core.serializers import UserSessionSerializer
+from rest_framework.permissions import IsAuthenticated
+
+
+class UserSessionViewSet(viewsets.ViewSet):
+    permission_classes = [IsAuthenticated]
+
+    def retrieve(self, request):
+        session, created = UserSession.objects.get_or_create(
+            user=request.user
+        )
+        serializer = UserSessionSerializer(session)
+        return Response(serializer.data)
+
+    def update(self, request):
+        session, created = UserSession.objects.get_or_create(
+            user=request.user
+        )
+
+        data = request.data
+
+        if 'analysisState' in data:
```

Comment:
you can just use get method here https://www.geeksforgeeks.org/python-dictionary-get-method/.

session.analysis_state = data.get('analysisState', '')